tests/playwright/src/model/pages/create-kind-cluster-page.ts

```diff
@@ -0,0 +1,115 @@
+    this.providerTypeCombobox = this.clusterPropertiesInformation.getByRole('combobox', { name: 'Provider Type' });
+    this.httpPort = this.clusterPropertiesInformation.getByLabel('HTTP Port');
+    this.httpsPort = this.clusterPropertiesInformation.getByLabel('HTTPS Port');
+    this.containerImage = this.clusterPropertiesInformation.getByPlaceholder('Leave empty for using latest.');
+    this.goBackButton = this.page.getByRole('button', { name: 'Go back to resources' });
+  }
+
+  public async createClusterDefault(clusterName: string, timeout?: number): Promise<void> {
+    await this.fillTextbox(this.clusterNameField, clusterName);
+    await playExpect(this.providerTypeCombobox).toHaveValue('podman');
+    await playExpect(this.httpPort).toHaveValue('9090');
+    await playExpect(this.httpsPort).toHaveValue('9443');
+    await playExpect(this.controllerCheckbox).toBeChecked();
+    await playExpect(this.containerImage).toBeEmpty();
+    await this.createCluster(timeout);
+  }
+
+  public async createClusterParametrized(
+    { clusterName, providerType, httpPort, httpsPort, useIngressController, containerImage }: KindClusterOptions = {},
+    timeout?: number,
+  ): Promise<void> {
+    if (clusterName) {
+      await this.fillTextbox(this.clusterNameField, clusterName);
+    }
+
+    if (providerType) {
+      await playExpect(this.providerTypeCombobox).toBeVisible();
+      const providerTypeOptions = await this.providerTypeCombobox.locator('option').allInnerTexts();
+      if (providerTypeOptions.includes(providerType)) {
+        await this.providerTypeCombobox.selectOption({ value: providerType });
+        await playExpect(this.providerTypeCombobox).toHaveValue(providerType);
+      } else {
+        throw new Error(`${providerType} doesn't exist`);
+      }
+
+      if (httpPort) {
+        await this.fillTextbox(this.httpPort, httpPort);
+      }
+      if (httpsPort) {
+        await this.fillTextbox(this.httpsPort, httpsPort);
+      }
+
+      if (!useIngressController) {
+        await playExpect(this.controllerCheckbox).toBeEnabled();
+        await this.controllerCheckbox.uncheck();
+        await playExpect(this.controllerCheckbox).not.toBeChecked();
+      }
+
+      if (containerImage) {
+        await this.fillTextbox(this.containerImage, containerImage);
+      }
+      await this.createCluster(timeout);
+    }
+  }
+
+  private async createCluster(timeout: number = 120000): Promise<void> {
+    await playExpect(this.clusterCreationButton).toBeVisible();
+    await this.clusterCreationButton.click();
+    await playExpect(this.goBackButton).toBeVisible({ timeout: timeout });
+    await this.goBackButton.click();
+  }
+
+  private async fillTextbox(textbox: Locator, text: string): Promise<void> {
+    await playExpect(textbox).toBeVisible();
+    await textbox.fill(text);
+    await playExpect(textbox).toHaveValue(text);
+  }
```

Comment:
Last week, I was looking for an issue that I could work on and noticed that we have at least two issues related to the Kubernetes cluster. Perhaps we should move the functions for creating a Kind cluster and checking if the Kind CLI is installed into the operations? Or would it be better to do this in a separate PR? WDYT @cbr7  @odockal?

Reply:
Yes, I agree, if there is a functionality that will be shared eventually, do it as part of another PR.